app/scripts/controllers/permissions/specifications.js

```diff
@@ -33,58 +33,29 @@ export const PermissionNames = Object.freeze({
  * PermissionController.
  */
 export const CaveatFactories = Object.freeze({
-  [CaveatTypes.restrictReturnedAccounts]: (accounts) => {
-    return { type: CaveatTypes.restrictReturnedAccounts, value: accounts };
-  },
-
-  [CaveatTypes.restrictNetworkSwitching]: (chainIds) => {
-    return { type: CaveatTypes.restrictNetworkSwitching, value: chainIds };
-  },
+  [Caip25CaveatType]: createCaip25Caveat,
 });
 
 /**
  * Gets the specifications for all caveats that will be recognized by the
  * PermissionController.
  *
- * @param {{
- *   getInternalAccounts: () => Record<string, import('@metamask/keyring-api').InternalAccount>,
- * }} options - Options bag.
+ * @param options - The options object.
+ * @param options.listAccounts - A function that returns the
+ * `AccountsController` internalAccount objects for all evm accounts.
+ * @param options.findNetworkClientIdByChainId - A function that
+ * returns the networkClientId given a chainId.
+ * @returns the caveat specifications to construct the PermissionController.
  */
 export const getCaveatSpecifications = ({
-  getInternalAccounts,
+  listAccounts,
   findNetworkClientIdByChainId,
 }) => {
   return {
-    [CaveatTypes.restrictReturnedAccounts]: {
-      type: CaveatTypes.restrictReturnedAccounts,
-
-      decorator: (method, caveat) => {
-        return async (args) => {
-          const result = await method(args);
-          return result.filter((account) => caveat.value.includes(account));
-        };
-      },
-
-      validator: (caveat, _origin, _target) =>
-        validateCaveatAccounts(caveat.value, getInternalAccounts),
-
-      merger: (leftValue, rightValue) => {
-        const newValue = Array.from(new Set([...leftValue, ...rightValue]));
-        const diff = newValue.filter((value) => !leftValue.includes(value));
-        return [newValue, diff];
-      },
-    },
-    [CaveatTypes.restrictNetworkSwitching]: {
-      type: CaveatTypes.restrictNetworkSwitching,
-      validator: (caveat, _origin, _target) =>
-        validateCaveatNetworks(caveat.value, findNetworkClientIdByChainId),
-      merger: (leftValue, rightValue) => {
-        const newValue = Array.from(new Set([...leftValue, ...rightValue]));
-        const diff = newValue.filter((value) => !leftValue.includes(value));
-        return [newValue, diff];
-      },
-    },
-
+    [Caip25CaveatType]: caip25CaveatBuilder({
```

Comment:
Are we still supporting merging in the new caveat?

Reply:
Ticket here: https://app.zenhub.com/workspaces/wallet-api-platform-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/3833